### ISSUE TITLE
how to check the "backward pass" is in "expanded space" in your code? 

### ISSUE THREAD

**Issue by aoaforever** [ARM-software/sesr#4]
Hello, wonderful for your work! I wish to know: how to check the "backward pass" is in "expanded space" in your code? 
I usually use the pytorch, so i know little about the tensorflow. Could you please tell me the answer？

**Comment by kbhardwa** [comment]
Hello. One easy way to do this is to look at the number of trainable parameters in the `--linear_block_type=expanded` experiment vs. `--linear_block_type=collapsed` experiment. We have now included a `model.summary()` statement in [train.py](https://github.com/ARM-software/sesr/blob/4813fa1e9ece28cea5abc6f55c6738da7c97cd02/train.py#L122). At the end of training, it prints out number of trainable parameters. You can easily see that the number of trainable parameters are exactly the same between "expanded linear block" experiment and "collapsed linear block" experiment (and is much larger than a model without any linear blocks). 

A more detailed way would be to look at model weights/gradients. This can be accomplished using keras callbacks. Please see a potential solution [here](https://stackoverflow.com/questions/57759635/get-gradients-with-keras-tensorflow-2-0).

**Comment by aoaforever** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
